agent/lib/kontena/websocket_client.rb

```diff
@@ -126,10 +129,20 @@ def send_request(id, method, params)
     # @param [Faye::WebSocket::API::Event] event
     def on_open(event)
       ping_timer.cancel if ping_timer
-      info 'connection established'
-      @connected = true
+
+      server_version = ws.headers['Kontena-Version']
+
       @connecting = false
-      notify_actors('websocket:open', event)
+      connecting_dt = Time.now - @connecting_at
+      @connected = true
+      @connected_at = Time.parse(@ws.headers['Kontena-Connected-At'])
+      connected_dt = Time.now - @connected_at
+
+      info "connected to master #{@api_uri} server version #{server_version} in #{'%.2fs' % connecting_dt} at #{@connected_at}, #{'%.2fs' % connected_dt} ago"
```

Comment:
The `@connecting_at` is the agent timestamp, and the `@connected_at` is the server timestamp, so this attempts to compare the two to log any clock offsets.